### PR TITLE
scripts: add update script.

### PR DESCRIPTION
**Pull request by siddontang**
@shenli @queenypingcap

### CODE REVIEW

File: op-guide/build.md [hunk 1]
```diff
@@ -23,3 +23,4 @@ install the missing ones automatically.
 
 You can use the [build script](../scripts/build.sh) to build and install TiDB components in the `bin` directory.
 
+You can use the [update script](../scripts/update.sh) to update the newest version after building before.
```
Comment (queenypingcap) [inline op-guide/build.md]: update the newest version after building before.
->
to update to the latest version.

File: op-guide/build.md [hunk 2]
```diff
@@ -23,3 +23,4 @@ install the missing ones automatically.
 
 You can use the [build script](../scripts/build.sh) to build and install TiDB components in the `bin` directory.
 
+You can use the [update script](../scripts/update.sh) to update to the latest version.
```
Comment (shenli) [inline op-guide/build.md]: update all the TiDB components to the latest version

File: scripts/update.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+TIDB_PATH=`pwd`
+echo "updating and building TiDB components in $TIDB_PATH"
+
+# All the binaries are installed in the `bin` directory. 
+mkdir -p $TIDB_PATH/bin
+
+# Assume we install go in /usr/local/go
+export PATH=$PATH:/usr/local/go/bin
+
+echo "checking go is installed"
```
Comment (shenli) [inline scripts/update.sh]: checking if go is installed
So as the following comments.
@queenypingcap PTAL

File: scripts/update.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# go is required
```
Comment (queenypingcap) [inline scripts/update.sh]: go ->Go

File: scripts/update.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+go version 
+# go version go1.6 darwin/amd64
```
Comment (queenypingcap) [inline scripts/update.sh]: +# The output might be like: go version go1.6 darwin/amd64

File: scripts/update.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+echo "checking rust is installed"
+# rust nightly is required
```
Comment (queenypingcap) [inline scripts/update.sh]: rust ->Rust

File: scripts/update.sh [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+rustc -V
+# rustc 1.12.0-nightly (7ad125c4e 2016-07-11)
```
Comment (queenypingcap) [inline scripts/update.sh]: +# The output might be like: rustc 1.12.0-nightly (7ad125c4e 2016-07-11)

File: scripts/update.sh [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+# GOPATH should be set correctly.
```
Comment (queenypingcap) [inline scripts/update.sh]: +# Set the GOPATH correctly.

File: scripts/update.sh [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+export GOPATH=$TIDB_PATH/deps/go
+
+# build TiDB
```
Comment (queenypingcap) [inline scripts/update.sh]: build -> Build

File: scripts/update.sh [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+echo "updating and building TiDB..."
+cd $GOPATH/src/github.com/pingcap/tidb
+git pull
+
+make
+cp -f ./bin/tidb-server $TIDB_PATH/bin
+cd $TIDB_PATH
+echo "build TiDB OK"
```
Comment (queenypingcap) [inline scripts/update.sh]: "build TiDB OK" -> "TiDB is built."

File: scripts/update.sh [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+# build PD
+echo "updating and building PD..."
+cd $GOPATH/src/github.com/pingcap/pd
+
+make
+cp -f ./bin/pd-server $TIDB_PATH/bin
+cp -rf ./templates $TIDB_PATH/bin/templates
+cd $TIDB_PATH
+echo "build PD OK"
```
Comment (queenypingcap) [inline scripts/update.sh]: "build PD OK" -> "PD is built."

File: scripts/update.sh [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+echo "updating and building TiDB..."
+cd $GOPATH/src/github.com/pingcap/tidb
+git pull
+
+make
+cp -f ./bin/tidb-server $TIDB_PATH/bin
+cd $TIDB_PATH
+echo "build TiDB OK"
+
+# build PD
```
Comment (queenypingcap) [inline scripts/update.sh]: build ->Build

File: scripts/update.sh [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+echo "updating and building PD..."
+cd $GOPATH/src/github.com/pingcap/pd
+
+make
+cp -f ./bin/pd-server $TIDB_PATH/bin
+cp -rf ./templates $TIDB_PATH/bin/templates
+cd $TIDB_PATH
+echo "build PD OK"
+
+# build TiKV
```
Comment (queenypingcap) [inline scripts/update.sh]: build ->Build

File: scripts/update.sh [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+echo "updating and building TiKV..."
+cd $TIDB_PATH/deps/tikv
+git pull 
+
+ROCKSDB_SYS_STATIC=1 ROCSDB_SYS_PORTABLE=1 make
+
+cp -f ./bin/tikv-server $TIDB_PATH/bin
+cd $TIDB_PATH
+echo "build TiKV OK"
```
Comment (queenypingcap) [inline scripts/update.sh]: "build TiKV OK" -> "TiKV is built."

**Comment by shenli** [comment]
LGTM

**Pull request merged**